app/controllers/api/v1/search_controller.rb

```diff
@@ -493,6 +559,25 @@ def self.match_studies_by_facet(query_results, search_facets)
         end
         matches
       end
+
+      # find valid StudyAccessions from query parameters
+      # only returns accessions currently in use
+      def self.find_matching_accessions(raw_accessions)
+        accessions = split_query_param_on_delim(parameter: raw_accessions)
+        sanitized_accessions = StudyAccession.sanitize_accessions(accessions)
+        Study.where(:accession.in => sanitized_accessions).map(&:accession)
```

Comment:
in cases like this where you're just getting ids it's good to be in the habit of using pluck rather than map for performance and clarity reasons.  (although for this particular case it probably doesn't make much of a difference)